src/BaseDir.hs

```diff
@@ -0,0 +1,24 @@
+module BaseDir 
+    ( validateDir
+    , baseDirArg
+    )
+    where
+
+import Prologue
+
+import Options.Applicative
+import Path.IO (resolveDir', doesDirExist)
+import System.Exit (die)
+
+-- | Validate that a filepath points to a directory and the directory exists
+validateDir :: FilePath -> IO (Path Abs Dir)
+validateDir dir = do
+  absolute <- resolveDir' dir
+  exists <- doesDirExist absolute
+
+  unless exists (die $ "ERROR: Directory " <> show absolute <> " does not exist")
+
+  pure absolute
+
+baseDirArg :: Parser String
```

Comment:
to make the intent clearer -- and because `String` is so vague (and `FilePath` is just an alias for `String`) -- you can use a newtype here:

```hs
newtype Basedir = Basedir { unBasedir :: FilePath }
```

and then change the type of `validateDir`/`baseDirArg`:
```hs
validateBasedir :: Basedir -> IO (Path Abs Dir)
baseDirArg :: Parser Basedir
```